pandas/tests/indexes/test_category.py

```diff
@@ -507,6 +507,22 @@ def test_identical(self):
         self.assertTrue(ci1.identical(ci1.copy()))
         self.assertFalse(ci1.identical(ci2))
 
+    def test_ensure_copied_data(self):
+        # Check the "copy" argument of each Index.__new__ is honoured
+        # GH12309
+        # Must be tested separately from other indexes because
+        # self.value is not an ndarray
+        for index in self.indices.values():
+            result = CategoricalIndex(index.values, copy=True)
+            tm.assert_index_equal(index, result)
+            if index.values.base is not None:
+                self.assertIsNot(index.values.base, self.result.base)
+
+            result = CategoricalIndex(index.values, copy=False)
+            bases = [i.values if i.values.base is None else i.values.base
```

Comment:
I think ``assert_numpy_array_equal`` should allow ``None`` to compare for the ``check_copy`` flag then.

also add some tests in ``tests/test_testing.py`` for ``check_same``